lake/commits/snapshot.go

```diff
@@ -72,6 +72,16 @@ func (s *Snapshot) DeleteObject(id ksuid.KSUID) error {
 	}
 	delete(s.objects, id)
 	s.deletedObjects[id] = object
+	// Remove any indexes attached to object.
+	rules, err := s.LookupIndexObjectRules(id)
+	if err != nil && !errors.Is(err, ErrNotFound) {
+		return err
+	}
+	for _, rule := range rules {
+		if err := s.DeleteIndexObject(rule.RuleID(), id); err != nil {
+			return err
+		}
+	}
```

Comment:
When else would this happen?

Reply:
It also happens when you delete a data object, e.g. with `zed delete`.

Reply:
To be clear, my agenda here it twofold. 
1. Make sure we're doing something reasonable if an error occurs. (That might be the case right now, but it isn't obvious.)
2. Make sure this is well tested. (There's no `zed index apply` in the `zed compact` tests, and there's no testing of this for `zed delete`.)

Reply:
Alright I see, I did this in snapshot.go so this will also effect the delete command. I'll remove the index delete here and apply this in a follow up pr where this'll get properly tested for both compact and delete commands (since it appears we are all in favor of doing this on object deletes).